filter.go

```diff
@@ -8,21 +8,34 @@ import (
 	manet "github.com/multiformats/go-multiaddr-net"
 )
 
+type FilterEntry struct {
+	f      *net.IPNet
+	reject bool
+}
+
 type Filters struct {
-	mu      sync.RWMutex
-	filters map[string]*net.IPNet
+	mu            sync.RWMutex
+	filterDefault bool
+	filters       []*FilterEntry
```

Comment:
Ah, if this is about https://github.com/ipfs/go-ipfs/issues/1972#issuecomment-157632849 disregard this comment.